src/build/_compat.py

```diff
@@ -28,12 +28,15 @@ def wrapper(cls):  # type: (_T) -> _T
 if sys.version_info[0] == 2:
     abstractproperty = abc.abstractproperty
 else:
-    from typing import Any, cast
+    from typing import TYPE_CHECKING, Any, cast
 
-    F = TypeVar('F', bound=Callable[..., Any])
+    if TYPE_CHECKING:  # pragma: no cover
+        abstractproperty = property  # pragma: no cover
+    else:
+        F = TypeVar('F', bound=Callable[..., Any])
 
-    def abstractproperty(func):  # type: (F) -> F
-        return cast(F, property(abc.abstractmethod(func)))
+        def abstractproperty(func):  # type: (F) -> Any
+            return cast(Any, property(abc.abstractmethod(func)))
```

Comment:
The cast here is redundant if you are returning Any though this function is not type checked anyway.